middleware.ts

```diff
@@ -12,8 +16,5 @@ export function middleware(req: NextRequest) {
     }
   }
 
-  return new Response('Auth required', {
-    status: 401,
-    headers: { 'WWW-Authenticate': 'Basic realm="Secure Area"' },
-  });
+  return NextResponse.json({ message: 'Auth required' }, { status: 401 });
```

Comment:
I hate next and this is why:

- https://nextjs.org/docs/messages/returning-response-body-in-middleware
- https://nextjs.org/docs/messages/nested-middleware

It just broke the contract of its APIs in a minor release 🫠